### DISCUSSION TITLE
coupled_variables problem disappeared and then reurned...

### DISCUSSION THREAD

**Discussion by MScKarl** [idaholab/moose#23817] [Q&A Modules: Phase field] [answered]
I started writing a Multiphase system using both “args” and “couple_ variables” for my system. On the first try I used “args” and received the following error:

*** ERROR ***
f_eta: Coupled variable 'fa' was not found.

Believing I should use “couple_ variables” instead I switched, resulting in the following error:

*** ERROR ***
Invalid function
(eta1^2-1)^2
in ParsedMaterialHelper.
Syntax error: Unknown identifier

Yesterday I was in communication with this forum and got some great advice, and after rebooting the system it worked, but now the problem is back, and restarting again doesn’t help. Could use some advice/tips on how to proceed. 

The code for my materials section follows below:

```
[Materials]

  [./kappa]
    type = GenericConstantMaterial
    prop_names  = 'kappa11 kappa12 kappa21 kappa22'
    prop_values = '0.01    0.01    0.01    0.01   '
  [../]

  [./L]
    type = GenericConstantMaterial
    prop_names = 'L'
    prop_values = '1.0'
  [../]

  [./P0]
    type = GenericConstantMaterial
    prop_names = 'P0'
    prop_values = '144000000'
  [../]

  [./s]
    type = GenericConstantMaterial
    prop_names = 's'
    prop_values = '0.6' # variera
  [../]


  [./fa]
    type = DerivativeParsedMaterial
    property_name = fa
    coupled_variables = 'eta1'
    function = '(eta1^2-1)^2'
    enable_jit = true
  [../]

  [./fb]
    type = DerivativeParsedMaterial
    f_name = fb
    coupled_variables = 'eta2'
    function = '(eta2^2-1)^2'
  [../]

  [./fc]
    type = DerivativeParsedMaterial
    f_name = fc
    coupled_variables = 'eta1'
    constant_names = 's'
    constant_expressions = '0.6'
    function = 's/4*(eta1*(eta1+2)*(eta1^2+2*eta1-6)-7)'
  [../]

   [./switching1]
    type = SwitchingFunctionMaterial
    function_name = h1
    eta = eta1
    h_order = HIGH
  [../]

  [./switching2]
    type = SwitchingFunctionMaterial
    function_name = h2
    eta = eta2
    h_order = HIGH
  [../]


  [./barrier]
    type = MultiBarrierFunctionMaterial
    etas = 'eta1 eta2'
  [../]

  [./f_eta]
    type = DerivativeParsedMaterial
    f_name = f_eta # Name of the function
    coupled_variables = 'fa fb fc' 
    constant_names = 'P0'
    constant_expressions = '144*10^6'
    function = 'P0*(fa+fb*fc)'
  [../]

[]

**Comment by GiudGiud** [comment]
You can ONLY use coupled_variables if your MOOSE is up to date.

If not, you MUST use args.

But here, fa fb and fc are not variables. They are material properties. So this is not the right parameter to use. 
With an UP-TO-DATE MOOSE, you have to use material_property_names for fa fb fc
https://mooseframework.inl.gov/source/materials/ParsedMaterial.html

The fact that this is unclear is exactly why we renamed these parameters (in an up to date MOOSE)

**Reply by MScKarl** [reply]
I re-installed MOOSE 4 Jan 2023. But when was the up-to-date moose published and can I get access to it without re-installing MOOSE again?
git log:

```
commit 66833b334e04e1e41d99663ecf3331f2530be536 (HEAD -> master)
Author: Your Name <you@example.com>
Date:   Wed Jan 4 23:12:07 2023 +0100
    initial commit

**Reply by GiudGiud** [reply]
the renaming was merged end of January. But today's is always the best version

**Reply by GiudGiud** [reply]
this page has how to udpate.
You ll likely need to update libmesh too
https://mooseframework.inl.gov/getting_started/installation/index.html

**Reply by MScKarl** [reply]
I will update and see if I can get it to work. There is a chance I may have a follow up question later.

**Reply by GiudGiud** [reply]
Sounds good.
btw what you posted and deleted is the same problem
```
  [./fa]
    type = DerivativeParsedMaterial
    property_name = fa
    coupled_variables = 'eta1'
    function = '(eta1^2-1)^2'
    enable_jit = true
  [../]
```
is not valid with an old moose. You need to say `f_name = fa` with an OLD moose. Current one, property_name is good

**Comment by MScKarl** [comment, marked as answer]
Hi again!

Now I've updated MOOSE and corrected the syntax, but I still get an error. 

*** ERROR ***
The following error occurred in the object "MOOSE Problem", of type "FEProblem".

Material property 'eta1', requested by 'fa' is not defined on block 0
Material property 'eta1', requested by 'fa_face' is not defined on block 0
Material property 'eta1', requested by 'fa_neighbor' is not defined on block 0
Material property 'eta2', requested by 'fb' is not defined on block 0
Material property 'eta2', requested by 'fb_face' is not defined on block 0
Material property 'eta2', requested by 'fb_neighbor' is not defined on block 0
Material property 'eta1', requested by 'fc' is not defined on block 0
Material property 'eta1', requested by 'fc_face' is not defined on block 0
Material property 'eta1', requested by 'fc_neighbor' is not defined on block 0


```
[Materials]

  [./kappa]
    type = GenericConstantMaterial
    prop_names  = 'kappa11 kappa12 kappa21 kappa22'
    prop_values = '0.01    0.01    0.01    0.01   '
  [../]

  [./L]
    type = GenericConstantMaterial
    prop_names = 'L'
    prop_values = '1.0'
  [../]

  [./P0]
    type = GenericConstantMaterial
    prop_names = 'P0'
    prop_values = '144000000'
  [../]

  [./s]
    type = GenericConstantMaterial
    prop_names = 's'
    prop_values = '0.6' # variera
  [../]


  [./fa]
    type = DerivativeParsedMaterial
    property_name = fa
    material_property_names = 'eta1'
    expression= '(eta1^2-1)^2'
    enable_jit = true
  [../]

  [./fb]
    type = DerivativeParsedMaterial
    property_name = fb
    material_property_names = 'eta2'
    expression= '(eta2^2-1)^2'
  [../]

  [./fc]
    type = DerivativeParsedMaterial
    property_name = fc
    material_property_names = 'eta1'
    constant_names = 's'
    constant_expressions = '0.6'
    expression= 's/4*(eta1*(eta1+2)*(eta1^2+2*eta1-6)-7)'
  [../]

   [./switching1]
    type = SwitchingFunctionMaterial
    function_name = h1
    eta = eta1
    h_order = HIGH
  [../]

  [./switching2]
    type = SwitchingFunctionMaterial
    function_name = h2
    eta = eta2
    h_order = HIGH
  [../]


  [./barrier]
    type = MultiBarrierFunctionMaterial
    etas = 'eta1 eta2'
  [../]

  [./f_eta]
    type = DerivativeParsedMaterial
    property_name = f_eta # Name of the function
    material_property_names = 'fa fb fc' 
    constant_names = 'P0'
    constant_expressions = '144*10^6'
    expression= 'P0*(fa+fb*fc)'
  [../]

[]

**Reply by GiudGiud** [reply]
So eta1 2 and 3 are variables right?

so they should be coupled_variables not material_property_names in fa fb fc

**Reply by MScKarl** [reply]
Thank you very much!

(Removed the other comment because it became moot.)
